### ISSUE TITLE
refactor: separate quote hooks per feature

### ISSUE THREAD

**Issue by janndriessen** [IndexCoop/index-app#1331]
Check if we can find a way to separate the individual quote hooks more and then let features have there individual mechanisms for getting quotes. As the requirement for features like leverage suite or trade are very different it would be better to split this up (even if repeating some code). This way it's easier customizable to the individual needs.

### Progress
Some of this has happened already e.g. legacy has its own quote functions.

## TODOs
- [ ] first separate custom logic of different features e.g. only swap or only leverage suite to their appropriate folders

**Comment by janndriessen** [comment]
Thoughts on this? @0xonramp

**Comment by janndriessen** [comment]
A lot of this might change with recent token deprecations. So let's wait on decisions here.

**Comment by janndriessen** [comment]
With `useQuoteResult` we now have a common hook for earn and leverage. `useBestQuote` will be removed once `Swap` is no longer needed.
